db/migrate/20211027041647_create_common_exposure_cohorts.rb

```diff
@@ -0,0 +1,245 @@
+        Patient.where(purged: false).where_assoc_exists(:common_exposure_cohorts).in_batches(of: PATIENT_BATCH_SIZE).each do |batch_group|
+          updates = {}
+          CommonExposureCohort.where(patient_id: batch_group.pluck(:id)).order(:updated_at).each do |common_exposure_cohort|
+            updates[common_exposure_cohort[:patient_id]] = {
+              member_of_a_common_exposure_cohort_type: common_exposure_cohort.cohort_name ||
+                                                       common_exposure_cohort.cohort_type ||
+                                                       common_exposure_cohort.cohort_location
+            }
+          end
+          Patient.update(updates.keys, updates.values)
+        end
+
+        # Rollback saved advanced filters
+        UserFilter.where('contents LIKE "%common-exposure-cohort%"').in_batches(of: FILTER_BATCH_SIZE) do |batch|
+          batch.each do |uf|
+            contents = JSON.parse(uf[:contents])
+            rollback_advanced_filter_contents(contents)
+            uf.update!(contents: contents.to_json)
+          end
+        end
+
+        # Rollback advanced filters in saved export presets
+        UserExportPreset.where('config LIKE "%common-exposure-cohort%"').in_batches(of: FILTER_BATCH_SIZE) do |batch|
+          batch.each do |uep|
+            config = JSON.parse(uep[:config])
+            contents = config.dig('data', 'patients', 'query', 'filter')
+            rollback_advanced_filter_contents(contents)
+            config['data']['patients']['query']['filter'] = contents
+            uep.update!(config: config.to_json)
+          end
+        end
+
+        # Rollback selected common exposure cohorts in saved export presets
+        UserExportPreset.where('config LIKE "%common_exposure_cohorts%"').in_batches(of: FILTER_BATCH_SIZE) do |batch|
+          batch.each do |uep|
+            config = JSON.parse(uep[:config])
+            if (config['data'].include?('common_exposure_cohorts'))
+              checked = config.dig('data', 'patients', 'checked')
+              checked.push('member_of_a_common_exposure_cohort_type')
+              config['data']['patients']['checked'] = checked
+              config['data'].delete('common_exposure_cohorts')
+              uep.update!(config: config.to_json)
+            end
+          end
+        end
+      end
+    rescue StandardError => e
+      puts 'An error has occured during the rollback, reverting changes...'
+      puts e
+
+      remove_column :patients, :member_of_a_common_exposure_cohort_type
+
+      # Raise another error to fail the migration
+      raise StandardError
+    else
+      drop_table :common_exposure_cohorts
+    ensure
+      ActiveRecord::Base.record_timestamps = true
+    end
+  end
+
+  def autopopulate_boolean_risk_factors
+    execute <<-SQL.squish
+      UPDATE patients
+      SET contact_of_known_case = true
+      WHERE purged = false AND contact_of_known_case_id IS NOT NULL
```

Comment:
What about empty string case for this and the similar queries below?

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1125/commits/70dc9e9a6374d1f1eb7dedccd85d1334a6e89afb